btf/types.go

```diff
@@ -1033,11 +1056,11 @@ func inflateRawTypes(rawTypes []rawType, rawStrings *stringTable) ([]Type, error
 
 	for _, fixup := range fixups {
 		i := int(fixup.id)
-		if i >= len(types) {
+		if i >= len(types)+len(baseTypes) {
```

Comment:
Can you bail out `if i < len(baseTypes)` here? Otherwise the interaction between `fixup()` and the `types[i-len(baseTypes)]` is pretty subtle.

Reply:
The loop here relies on the fact that `fixup` will unconditionally take care of base types, so we can never see an id < len(baseTypes). That isn't easily apparent from the code, adding `if i < len(baseTypes)` could serve as an indication for this. The alternative is a comment, but that is less robust to changes in `fixup`.